### PR TITLE
add nodejs based healthcheck for docker health management

### PR DESCRIPTION
**Pull request by bitofant**
added nodejs based health check for docker container health management. Healthcheck.js contains commented hints on how to extend status page parsing. The current implementation is based solely on the http status code.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -12,6 +12,8 @@ RUN npm install --production
 
 COPY . /src
 
+HEALTHCHECK --interval=10s --timeout=10s --start-period=5s --retries=3 CMD node healthcheck.js || exit 1
```
Comment (MartinLang1) [inline Dockerfile]: Why do we use different timings than in amass scanner?

(Same comment than amass PR:)
Timeout seems to be too long. I would suggest to use maybe 3 to 5 seconds. It should be at least shorter than the interval...

File: healthcheck.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+	response.on ('data', chunk => {
+		dataChunks.push (chunk);
+	});
+
+	response.on ('close', () => {
+		const txtResponse = Buffer.concat (dataChunks).toString ('utf8');
+		const jsonResponse = JSON.parse (txtResponse);
+		if (jsonResponse.healthcheck === 'DOWN') process.exit (1);
+	});
+	*/
```
Comment (MartinLang1) [inline healthcheck.js]: remove commented out code

File: Dockerfile (hunk 1 above)
Comment (bitofant) [reply Dockerfile]: all scanners now have the same 5s interval

File: healthcheck.js (hunk 2 above)
Comment (bitofant) [reply healthcheck.js]: file improved according to suggestions

**Pull request merged**